### PR TITLE
Grid Component Added

### PR DESCRIPTION
**Pull request by Pooran8898**
### NEW FEATURE

User can pass Grid Items as a Children to the Grid Component and basis on the props you define column count for different screen sizes it will render the children inside that grid container. 

Props
style: React.CssProperties
onGridContainerClick: React.MouseEventHandler
columns:{
mobile: number,
tablet: number,
desktop: number
}
overflow:{
mobile: number,
}
customWidth:[{
columnNumber: 1;
customWidth: number;
}]

Screenshot 2023-03-10 at 4 10 42 PM

Screenshot 2023-03-10 at 4 10 52 PM

Screenshot 2023-03-10 at 5 34 20 PM

### CODE REVIEW

File: src/components/Grid/Grid.interfaces.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  onGridContainerClick?: React.MouseEventHandler;
+  columns: {
+    [ScreenTypes.MOBILE]?: 1 | 2 | 3 | 4 | 6;
+    [ScreenTypes.TABLET]?: 1 | 2 | 3 | 4 | 6;
+    [ScreenTypes.DESKTOP]: 1 | 2 | 3 | 4 | 6;
+  };
+  overflow?: {
+    [ScreenTypes.MOBILE]: boolean;
+  };
+  // justifyContent?: {
```
Comment (ShahRohan27) [inline src/components/Grid/Grid.interfaces.ts:24]: Remove if not needed

File: src/components/Grid/Grid.scss [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+.n-grid-block-4 {
+  grid-template-columns: repeat(4, 1fr);
+}
+
+.n-grid-block-6 {
+  grid-template-columns: repeat(6, 1fr);
+}
+
+// If we want 8 column or 10 colum or 12 we can just add a new class here and it will automatically get added
```
Comment (ShahRohan27) [inline src/components/Grid/Grid.scss:49]: This doesnt seem dynamic. We cannot release new versions for support of extra columns. Please move this logic to JS and pass it as a style prop to the division

File: src/components/Grid/Grid.scss [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+.testing-child-container {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  font-size: 2rem;
+  color: $TypographyPrimaryColor;
+  background-color: rgb(187, 208, 226);
+  padding: 10px;
```
Comment (ShahRohan27) [inline src/components/Grid/Grid.scss:58]: Please use rem

**Review by ShahRohan27 — CHANGES_REQUESTED** [review]
Please check

File: src/components/Grid/Grid.interfaces.ts (hunk 1 above)
Comment (Pooran8898) [reply src/components/Grid/Grid.interfaces.ts]: Removed @ShahRohan27

File: src/components/Grid/Grid.scss (hunk 2 above)
Comment (Pooran8898) [reply src/components/Grid/Grid.scss]: @ShahRohan27  Yes now the column is also generating on js part

File: src/components/Grid/Grid.scss (hunk 3 above)
Comment (Pooran8898) [reply src/components/Grid/Grid.scss]: @ShahRohan27  Rem is also updated

File: src/components/Grid/Grid.scss (hunk 2 above)
Comment (ShahRohan27) [reply src/components/Grid/Grid.scss]: Then we should consider removing this classes, what will happen in case of 7 children ?

Comment (Pooran8898) [reply src/components/Grid/Grid.scss]: @ShahRohan27  All the classes are removed

**Comment by ShahRohan27** [comment]
LGTM 👍

**Pull request merged**